otter/cloud_client/__init__.py

```diff
@@ -705,21 +708,45 @@ def get_clbs():
         success=lambda (response, body): body['loadBalancers'])
 
 
-def get_clb_node_feed(lb_id, node_id):
-    """Get the atom feed associated with a CLB node. Returns feed as str."""
+def _node_feed_request(lb_id, node_id, params):
     return service_request(
-        ServiceType.CLOUD_LOAD_BALANCERS,
-        'GET',
+        ServiceType.CLOUD_LOAD_BALANCERS, 'GET',
         append_segments('loadbalancers', str(lb_id), 'nodes',
                         '{}.atom'.format(node_id)),
+        params=params,
         json_response=False
     ).on(
         error=_only_json_api_errors(
             lambda c, b: _process_clb_api_error(c, b, lb_id))
     ).on(
         log_success_response('request-get-clb-node-feed', identity)
-    ).on(
-        success=lambda (response, body): body)
+    ).on(itemgetter(1))
+
+
+@do
+def get_clb_node_feed(lb_id, node_id):
+    """
+    Get the atom feed associated with a CLB node.
+
+    :param int lb_id: Cloud Load balancer ID
+    :param int node_id: Node ID of in loadbalancer node
+
+    :returns: ``list`` of atom entry :class:`Element`
+    :rtype: ``list``
+    """
+    all_entries = []
+    params = {}
+    while True:
+        feed_str = yield _node_feed_request(lb_id, node_id, params)
+        feed = atom.parse(feed_str)
+        entries = atom.entries(feed)
+        if entries == []:
+            yield do_return(all_entries)
+        all_entries.extend(entries)
+        next_link = atom.next_link(feed)
+        if not next_link:
+            yield do_return(all_entries)
+        params = parse_qs(urlparse(next_link).query)
```

Comment:
I agree with you completely. However, this function works/returns with `Effect of ServiceRequest` which is tied to a specific service whose URI gets expanded during performing. It will get complicated to support any URI _with_ authentication. 

I had similar confusion when implementing getting all nova servers. But settled with a similar solution. But you are right, this should be documented as a tech-debt. Will do. Thanks!